StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -165,51 +165,73 @@ extension PaymentSheet {
+            let linkCardBrandAvailability = PaymentSheet.PaymentMethodType.supportsLinkCardIntegration(
+                configuration: configuration,
+                intent: intent,
+                elementsSession: elementsSession
+            )
+
+            let eligibleForInstantDebits: Bool
+            switch instantBankPaymentsAvailability {
+            case .supported:
+                eligibleForInstantDebits = true
+            case .primaryRequirementMetButMissingOtherRequirements:
+                if logAvailability {
+                    #if DEBUG
+                    print("[Stripe SDK]: PaymentSheet could not offer Instant Bank Payments: \(instantBankPaymentsAvailability.debugDescription)")
+                    #endif
+                }
+                fallthrough
+            default:
+                eligibleForInstantDebits = false
             }
 
-            if eligibleForInstantDebits {
-                let availabilityStatus = configurationSatisfiesRequirements(
-                    requirements: [.financialConnectionsSDK],
-                    configuration: configuration,
-                    intent: intent
-                )
-                if availabilityStatus == .supported {
-                    recommendedPaymentMethodTypes.append(.instantDebits)
+            let eligibleForLinkCardBrand: Bool
+            switch linkCardBrandAvailability {
+            case .supported:
+                eligibleForLinkCardBrand = true
+
+                // Instant Bank Payments take precedence over the Link Card integration,
+                // so this payment won't be shown even though all other requirements are met.
+                if eligibleForInstantDebits, logAvailability {
+                    #if DEBUG
+                    print("[Stripe SDK]: PaymentSheet could not offer the Link Card integration: This configuration supports Instant Bank Payments, which takes precedence over the Link Card integration")
+                    #endif
                 }
-            // Else if here so we don't show both Instant Debits and Link Card Brand together.
-            } else if eligibleForLinkCardBrand {
-                let availabilityStatus = configurationSatisfiesRequirements(
-                    requirements: [.financialConnectionsSDK],
-                    configuration: configuration,
-                    intent: intent
-                )
-                if availabilityStatus == .supported {
-                    recommendedPaymentMethodTypes.append(.linkCardBrand)
+            case .primaryRequirementMetButMissingOtherRequirements:
+                if logAvailability {
+                    #if DEBUG
+                    print("[Stripe SDK]: PaymentSheet could not offer the Link Card integration: \(linkCardBrandAvailability.debugDescription)")
+                    #endif
                 }
+                fallthrough
+            default:
+                eligibleForLinkCardBrand = false
```

Comment:
Should this log `[Stripe SDK]: PaymentSheet is not adding the 'Bank' payment method type: \(description)`?

Reply:
The primary requirement is not met in this scenario, so we don't log anything here